File: Assignment35_4.c
#include <stdio.h>
#include <stdbool.h>             // Header for bool datatype

///////////////////////////////////////////////////////////////////////////////
// Function Name : ChkVowel
// Description   : Check if the string contains at least one vowel (a,e,i,o,u)
// Input         : String
// Output        : TRUE or FALSE
///////////////////////////////////////////////////////////////////////////////
bool ChkVowel(char *str)
{
    while(*str != '\0')          // Traverse string
    {   // Check for lowercase or uppercase vowels
        if(*str=='a' || *str=='e' || *str=='i' || *str=='o' || *str=='u' ||
           *str=='A' || *str=='E' || *str=='I' || *str=='O' || *str=='U')
        {
            return true;         // Vowel found → TRUE
        }
        str++;                   // Move pointer
    }
    return false;                // No vowel found → FALSE
}

int main()
{
    char arr[20];
    bool bRet = false;

    printf("Enter string : ");
    scanf("%[^'\n']s",arr);

    bRet = ChkVowel(arr);

    if(bRet == true)
    {
        printf("Contains Vowel\n");
    }
    else
    {
        printf("There is no Vowel\n");
    }

    return 0;
}

File: Assignment35_4.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : ChkVowel
// Description   : Check if the string contains vowels (a,e,i,o,u)
// Input         : String
// Output        : TRUE or FALSE
// Author        : Raj Samir Jadhav
// Date          : 03/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    public boolean ChkVowel(String str)     // Function
    {
        for(int i = 0; i < str.length(); i++)
        {
            char ch = str.charAt(i);

            // Check lowercase & uppercase vowels
            if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u' ||
               ch=='A' || ch=='E' || ch=='I' || ch=='O' || ch=='U')
            {
                return true;              // Vowel found
            }
        }
        return false;                     // No vowel found
    }
}

class Assignment35_4
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string : ");
        String input = sobj.nextLine();

        Pattern pobj = new Pattern();
        boolean bRet = pobj.ChkVowel(input);

        if(bRet == true)
        {
            System.out.println("Contains Vowel");
        }
        else
        {
            System.out.println("There is no Vowel");
        }

        sobj.close();                   // To avoid scanner warning
    }
}
